napari/components/_viewer/view/buttons.py

```diff
@@ -7,14 +7,14 @@
 
 
 class QtLayersButtons(QFrame):
-    def __init__(self, layers):
+    def __init__(self, viewer):
         super().__init__()
 
-        self.layers = layers
-        self.deleteButton = QtDeleteButton(self.layers)
-        self.newMarkersButton = QtNewMarkersButton(self.layers)
-        self.newShapesButton = QtNewShapesButton(self.layers)
-        self.newLabelsButton = QtNewLabelsButton(self.layers)
+        self.viewer = viewer
+        self.deleteButton = QtDeleteButton(self.viewer)
+        self.newMarkersButton = QtNewMarkersButton(self.viewer)
+        self.newShapesButton = QtNewShapesButton(self.viewer)
+        self.newLabelsButton = QtNewLabelsButton(self.viewer)
```

Comment:
Ah, this is what you meant in your PR comment. Yes, actually I do think it should probably go into the Layers view instead of here. Intuitively, it should be possible to instantiate a layersview with a totally different viewer, and there is no reason (?) why these manipulations couldn't happen in that scenario.

Reply:
yup, so this would argue that all of the `add_*` methods should go on the layers list. Would we also expose them on the viewer too for convenience - so you could do `viewer.add_image` but that would just call `viewer.layers.add_image` where all the magic would happen? I'm ok with that idea - and that would then lead me to undo these changes. That might actually be worth doing in this PR. Curious what @kne42 thinks? What's your take @jni - I could give it a try

Reply:
ah I actually won't be able to do it in this PR as the `viewer` still needs to get appended onto the layer so I'd need to refactor everything to get this to work. I'd say we live with this for now, and then once we've refactored the individual layers and the viewer we can put these buttons back in the layers view

Reply:
Yeah that sounds good to me!